dotCMS/src/main/java/com/dotcms/graphql/DotGraphQLContext.java

```diff
@@ -1,23 +1,117 @@
+import graphql.kickstart.execution.context.DefaultGraphQLContext;
+import graphql.kickstart.servlet.context.GraphQLServletContext;
+import org.dataloader.DataLoaderRegistry;
 
+import javax.security.auth.Subject;
 import javax.servlet.http.HttpServletRequest;
 import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.Part;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
 
-import graphql.servlet.GraphQLContext;
-
-public class DotGraphQLContext extends GraphQLContext {
+public class DotGraphQLContext extends DefaultGraphQLContext implements
+        GraphQLServletContext {
 
+    private final HttpServletRequest httpServletRequest;
+    private final HttpServletResponse httpServletResponse;
     private final User user;
 
-    public DotGraphQLContext(final HttpServletRequest httpServletRequest,
-                             final HttpServletResponse httpServletResponse, final User user) {
-        super(httpServletRequest, httpServletResponse);
+    private DotGraphQLContext(DataLoaderRegistry dataLoaderRegistry, Subject subject, HttpServletRequest httpServletRequest,
+            HttpServletResponse httpServletResponse, User user) {
+        super(dataLoaderRegistry, subject);
+        this.httpServletRequest = httpServletRequest;
+        this.httpServletResponse = httpServletResponse;
         this.user = user;
     }
 
+    @Override
+    public HttpServletRequest getHttpServletRequest() {
+        return httpServletRequest;
+    }
+
+    @Override
+    public HttpServletResponse getHttpServletResponse() {
+        return httpServletResponse;
+    }
+
+    @Override
+    public List<Part> getFileParts() {
+        try {
+            return httpServletRequest.getParts().stream()
+                    .filter(part -> part.getContentType() != null)
+                    .collect(Collectors.toList());
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    @Override
+    public Map<String, List<Part>> getParts() {
+        try {
+            return httpServletRequest.getParts()
+                    .stream()
+                    .collect(Collectors.groupingBy(Part::getName));
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        }
+    }
+
     public User getUser() {
         return user;
     }
+
+    public static Builder createServletContext(DataLoaderRegistry registry, Subject subject) {
+        return new Builder(registry, subject);
+    }
+
+    public static Builder createServletContext() {
+        return new Builder(new DataLoaderRegistry(), null);
+    }
+
+    public static class Builder {
+        private HttpServletRequest httpServletRequest;
+        private HttpServletResponse httpServletResponse;
+        private DataLoaderRegistry dataLoaderRegistry;
+        private Subject subject;
+        private User user;
+
+        private Builder(DataLoaderRegistry dataLoaderRegistry, Subject subject) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'dataLoaderRegistry' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5685344)